core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1308,25 +1308,30 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
     }
   }
 
-  case class PartitionReassignment(partitionReassignment: Map[TopicAndPartition, Seq[Int]]) extends ControllerEvent {
+  case object PartitionReassignment extends ControllerEvent {
 
     def state = ControllerState.PartitionReassignment
 
     override def process(): Unit = {
       if (!isActive) return
-      val partitionsToBeReassigned = partitionReassignment.filterNot(p => controllerContext.partitionsBeingReassigned.contains(p._1))
-      partitionsToBeReassigned.foreach { partitionToBeReassigned =>
-        if(topicDeletionManager.isTopicQueuedUpForDeletion(partitionToBeReassigned._1.topic)) {
-          error("Skipping reassignment of partition %s for topic %s since it is currently being deleted"
-            .format(partitionToBeReassigned._1, partitionToBeReassigned._1.topic))
-          removePartitionFromReassignedPartitions(partitionToBeReassigned._1)
-        } else {
-          val context = ReassignedPartitionsContext(partitionToBeReassigned._2)
-          initiateReassignReplicasForTopicPartition(partitionToBeReassigned._1, context)
+      // We read the reassignment data fresh so that we don't need to maintain it in memory. While
+      // a reassignment is in progress, there can be potentially many reassignment events in the queue
+      // since the completion of every individual reassignment causes the reassignment path to be updated.
+      val reassignmentDataOpt = zkUtils.readDataMaybeNull(ZkUtils.ReassignPartitionsPath)._1
```

Comment:
You could just use `ZkUtils.getPartitionsBeingReassigned()`

Reply:
Ack. I looked for such a function initially and couldn't find it.